src/main/java/com/example/clone/service/PostService.java

```diff
@@ -0,0 +1,74 @@
+package com.example.clone.service;
+
+
+import com.example.clone.model.Post;
+import com.example.clone.model.UserInfo;
+import org.springframework.stereotype.Service;
+import com.example.clone.repository.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import com.example.clone.repository.UserInfoRepository;
+
+import java.util.*;
+
+@Service
+
+public class PostService {
+
+    private final PostRepository postRepository;
+    private final GetAuth getAuth;
+    private final UserInfoRepository userRepository;
```

Comment:
Don't depend on other repository, but on other service. We want to avoid coupling that is violating SRP.

https://www.freecodecamp.org/news/solid-principles-single-responsibility-principle-explained/